patches/tModLoader/Terraria/GameContent/TownNPCProfiles.cs.patch

```diff
@@ -0,0 +1,60 @@
+ 			{ 369, new Profiles.LegacyNPCProfile("Images/TownNPCs/Angler", 22) },
++			{ NPCID.ArmsDealer, new Profiles.AlternateLegacyNPCProfile("Images/NPC_19", NPCHeadID.ArmsDealer) },
+ 			{ 633, new Profiles.TransformableNPCProfile("Images/TownNPCs/BestiaryGirl", 26) },
+ 			{ 54, new Profiles.LegacyNPCProfile("Images/TownNPCs/Clothier", 7) },
+ 			{ 209, new Profiles.LegacyNPCProfile("Images/TownNPCs/Cyborg", 16) },
++			{ NPCID.DD2Bartender, new Profiles.AlternateLegacyNPCProfile("Images/NPC_550", NPCHeadID.DD2Bartender) },
+ 			{ 38, new Profiles.LegacyNPCProfile("Images/TownNPCs/Demolitionist", 4) },
++			{ NPCID.Dryad, new Profiles.AlternateLegacyNPCProfile("Images/NPC_20", NPCHeadID.Dryad) },
+ 			{ 207, new Profiles.LegacyNPCProfile("Images/TownNPCs/DyeTrader", 14) },
++			{ NPCID.GoblinTinkerer, new Profiles.AlternateLegacyNPCProfile("Images/NPC_107", NPCHeadID.GoblinTinkerer) },
+ 			{ 588, new Profiles.LegacyNPCProfile("Images/TownNPCs/Golfer", 25) },
++			{ NPCID.Guide, new Profiles.AlternateLegacyNPCProfile("Images/NPC_22", NPCHeadID.Guide) },
+ 			{ 124, new Profiles.LegacyNPCProfile("Images/TownNPCs/Mechanic", 8) },
+ 			{ 17, new Profiles.LegacyNPCProfile("Images/TownNPCs/Merchant", 2) },
+ 			{ 18, new Profiles.LegacyNPCProfile("Images/TownNPCs/Nurse", 3) },
+ 			{ 227, new Profiles.LegacyNPCProfile("Images/TownNPCs/Painter", 17) },
++			{ NPCID.PartyGirl, new Profiles.AlternateLegacyNPCProfile("Images/NPC_208", NPCHeadID.PartyGirl) },
+ 			{ 229, new Profiles.LegacyNPCProfile("Images/TownNPCs/Pirate", 19) },
++			{ 663, new Profiles.LegacyNPCProfile("Images/TownNPCs/Princess", 45) },
+ 			{ 142, new Profiles.LegacyNPCProfile("Images/TownNPCs/Santa", 11) },
+ 			{ 453, new Profiles.LegacyNPCProfile("Images/TownNPCs/SkeletonMerchant", -1) },
+ 			{ 178, new Profiles.LegacyNPCProfile("Images/TownNPCs/Steampunker", 13) },
+ 			{ 353, new Profiles.LegacyNPCProfile("Images/TownNPCs/Stylist", 20) },
+ 			{ 441, new Profiles.LegacyNPCProfile("Images/TownNPCs/TaxCollector", 23) },
+ 			{ 368, new Profiles.LegacyNPCProfile("Images/TownNPCs/TravelingMerchant", 21) },
++			{ NPCID.Truffle, new Profiles.AlternateLegacyNPCProfile("Images/NPC_160", NPCHeadID.Truffle) },
++			{ NPCID.WitchDoctor, new Profiles.AlternateLegacyNPCProfile("Images/NPC_228", NPCHeadID.WitchDoctor) },
+ 			{ 108, new Profiles.LegacyNPCProfile("Images/TownNPCs/Wizard", 10) },
+-			{ 663, new Profiles.LegacyNPCProfile("Images/TownNPCs/Princess", 45) },
+ 			{ 637, new Profiles.VariantNPCProfile("Images/TownNPCs/Cat", "Cat", CatHeadIDs, "Siamese", "Black", "OrangeTabby", "RussianBlue", "Silver", "White") },
+ 			{ 638, new Profiles.VariantNPCProfile("Images/TownNPCs/Dog", "Dog", DogHeadIDs, "Labrador", "PitBull", "Beagle", "Corgi", "Dalmation", "Husky") },
+ 			{ 656, new Profiles.VariantNPCProfile("Images/TownNPCs/Bunny", "Bunny", BunnyHeadIDs, "White", "Angora", "Dutch", "Flemish", "Lop", "Silver") }		};
++
++		public Dictionary<int, ITownNPCProfile> TownNPCProfileList;
```

Comment:
This probably should be localized instead of held as part of the instance. 
I'm not seeing any calls to it outside of the one method

Reply:
I had originally done that and then had second thoughts about it. good to see I was right the first time :HDfailure: